featuregate/featuregatetest/test_util.go

```diff
@@ -11,7 +11,9 @@ import (
 	"go.opentelemetry.io/collector/featuregate"
 )
 
-func setFeatureGateForTest(tb testing.TB, gate *featuregate.Gate, enabled bool) {
+// SetGate sets the value to the given gate. Also, it installs a cleanup function to restore
+// the gate to the initial value when the test is done.
+func SetGate(tb testing.TB, gate *featuregate.Gate, enabled bool) {
```

Comment:
Why do this instead of using a local registry for testing? I am a bit worried about encouraging this pattern, it modifies global state and it can be hard to reason about

Reply:
Any example where I can see what you envision?